### PR TITLE
[Enhancement] Control the writing of metadata, for task_runs

### LINKED ISSUE #21654: Crazy refreshing of materialized view causes FE to fail to start
### Steps to reproduce the behavior (Required)
1. Crazy refreshing materialized view

### Real behavior (Required)
2023-02-20 00:00:37,251 ERROR (leaderCheckpointer|655) [Checkpoint.replayAndGenerateGlobalStateMgrImage():214] Exception when generate new image file
java.lang.IllegalArgumentException: null
        at java.nio.CharBuffer.allocate(CharBuffer.java:334) ~[?:1.8.0_144]
        at java.nio.charset.CharsetDecoder.decode(CharsetDecoder.java:810) ~[?:1.8.0_144]
        at com.starrocks.common.io.Text.decode(Text.java:336) ~[starrocks-fe.jar:?]
        at com.starrocks.common.io.Text.decode(Text.java:315) ~[starrocks-fe.jar:?]
        at com.starrocks.common.io.Text.readString(Text.java:390) ~[starrocks-fe.jar:?]
        at com.starrocks.scheduler.TaskManager.loadTasks(TaskManager.java:396) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.loadImage(GlobalStateMgr.java:1183) ~[starrocks-fe.jar:?]
        at com.starrocks.leader.Checkpoint.replayAndGenerateGlobalStateMgrImage(Checkpoint.java:201) [starrocks-fe.jar:?]
        at com.starrocks.leader.Checkpoint.runAfterCatalogReady(Checkpoint.java:93) [starrocks-fe.jar:?]
        at com.starrocks.common.util.LeaderDaemon.runOneCycle(LeaderDaemon.java:60) [starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.4~2.5

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #21654

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [x] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/scheduler/TaskManager.java [hunk 1]
```diff
@@ -465,7 +465,23 @@ public long saveTasks(DataOutputStream dos, long checksum) throws IOException {
         checksum ^= data.tasks.size();
         data.runStatus = showTaskRunStatus(null);
         String s = GsonUtils.GSON.toJson(data);
-        Text.writeString(dos, s);
+        boolean retry = false;
+        try {
+            Text.writeString(dos, s);
+        } catch (IOException ex) {
+            if (ex.getMessage().contains("larger than 1GB")) {
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/TaskManager.java:472]: Better not depend on the error msg.

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 2]
```diff
@@ -275,13 +275,19 @@ public class Config extends ConfigBase {
...
 
     /**
      * for task run set expire time
      */
     @ConfField(mutable = true)
-    public static int task_runs_ttl_second = 3 * 24 * 3600;     // 3 day
+    public static int task_runs_ttl_second = 24 * 3600;     // 1 day
+
+    /**
+     * for task run set expire time
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:287]:
```suggestion
     * max history task num kept
```

**Review by gengjun-git — APPROVED** [review]

**Review by ABingHuang — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**